test/test-PreviewProvider.js

```diff
@@ -50,6 +50,19 @@ const gMockMetadataStore = {
       });
     }
     return items;
+  },
+  asyncExecuteQuery(query) {
+    // extract key from the query
+    let key = query.split("=")[1].trim().replace(/'/g, "");
```

Comment:
I'm curious how you ended up on this path of directly querying as this test shim is quite fragile if someone was expecting to do some other query. Is the performance of `asyncGetMetadataByCacheKey` undesirable or some other issue?